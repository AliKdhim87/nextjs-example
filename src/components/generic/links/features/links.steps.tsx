import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { defineFeature, loadFeature } from 'jest-cucumber'
import Links from '../index'

const feature = loadFeature('src/components/generic/links/features/links.feature')

defineFeature(feature, (test) => {
  test('Add array of link objects', ({ when, then }) => {
    const component = renderer.create(
      <Links
        items={[
          { path: '/', name: 'Home' },
          { path: '/about', name: 'About' }
        ]}
      />
    )

    when('I provide the link objects', () => {
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
    then('The links should be generated', () => {
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
