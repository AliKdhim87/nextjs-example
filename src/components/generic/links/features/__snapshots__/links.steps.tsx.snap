// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Links component Add array of link objects 1`] = `
<ul>
  <li>
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Home
    </a>
  </li>
  <li>
    <a
      href="/about"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      About
    </a>
  </li>
</ul>
`;

exports[`Links component Add array of link objects 2`] = `
<ul>
  <li>
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Home
    </a>
  </li>
  <li>
    <a
      href="/about"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      About
    </a>
  </li>
</ul>
`;
